### PR TITLE
support for many_users and many_groups flag in /users controlpanel

### PR DESCRIPTION
**Pull request by nileshgulia1**
…panels

### CODE REVIEW

File: src/components/manage/Controlpanels/Groups/GroupsControlpanel.jsx [hunk 1]
```diff
@@ -108,11 +109,15 @@ class GroupsControlpanel extends Component {
...
 
   fetchData = async () => {
+    await this.props.getControlpanel('usergroup');
     await this.props.listRoles();
-    await this.props.listGroups();
-    this.setState({
-      groupEntries: this.props.groups,
-    });
+    if (!this.props.many_groups) {
+      await this.props.listRoles();
```
Comment (ksuess) [inline src/components/manage/Controlpanels/Groups/GroupsControlpanel.jsx:115]: I think one `this.props.listRoles()` is enough and is OK also for many_groups

**Review by ksuess — CHANGES_REQUESTED** [review]
LGTM if you remove the superfluous dispatch of listRoles

**Comment by ksuess** [comment]
Idea for a follow up PR: I think it's worth thinking about a proper pagination, as now users are fetched completely and not batch per batch.

**Comment by ksuess** [comment]
And I think the trip via component state is not necessary anymore, as you integrated the roles info in restapi endpoint "users".

File: src/components/manage/Controlpanels/Groups/GroupsControlpanel.jsx (hunk 1 above)
Comment (nileshgulia1) [reply src/components/manage/Controlpanels/Groups/GroupsControlpanel.jsx]: Done @ksuess , I have also added cypress for relevant scenarios.

File: Makefile [hunk 2]
```diff
@@ -14,7 +14,7 @@ MAKEFLAGS+=--no-builtin-rules
 
 INSTANCE_PORT=8080
 DOCKER_IMAGE=plone/plone-backend:5.2.7
-KGS=plone.restapi==8.22.0 plone.volto==4.0.0a4 plone.rest==2.0.0a5 plone.app.iterate==4.0.2 plone.app.vocabularies==4.3.0
+KGS=plone.restapi==8.24.0 plone.volto==4.0.0a4 plone.rest==2.0.0a5 plone.app.iterate==4.0.2 plone.app.vocabularies==4.3.0
```
Comment (ksuess) [inline Makefile:17]:
```suggestion
KGS=plone.restapi==8.24.1 plone.volto==4.0.0a8 plone.rest==2.0.0a5 plone.app.iterate==4.0.3 plone.app.vocabularies==4.3.0
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,8 @@
 
 ### Feature
 
+- support for many_users and many_groups flag in user controlpanel @nileshgulia1
```
Comment (ksuess) [inline CHANGELOG.md:11]:
```suggestion
- support for many_users and many_groups flag in user controlpanel and group controlpanel @nileshgulia1
```

**Review by ksuess — CHANGES_REQUESTED** [review]

**Review by ksuess — APPROVED** [review]

**Comment by ksuess** [comment]
@nileshgulia1 I think my suggestion to update test conditions (versions) was not well thought. I realized now, that tests are running on Plone 5.2.7. So the set of versions may not fit. You may want to check with someone who is familiar with the test scenario of Volto.

**Comment by nileshgulia1** [comment]
> @nileshgulia1 I think my suggestion to update test conditions (versions) was not well thought. I realized now, that tests are running on Plone 5.2.7. So the set of versions may not fit. You may want to check with someone who is familiar with the test scenario of Volto.

I think we need this PR merged first. https://github.com/plone/volto/pull/3544

**Review by sneridagh — APPROVED** [review]

**Pull request merged**